python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -68,7 +75,17 @@ def get_nodes_to_launch(
             pending_nodes: Summary of node types currently being launched.
             resource_demands: Vector of resource demands from the scheduler.
             usage_by_ip: Mapping from ip to available resources.
+            pending_placement_groups: Placement group demands.
+            static_node_resources: Mapping from ip to static node resources.
         """
+        if self.is_legacy_yaml:
+            # When using legacy yaml files and after launching worker nodes we
+            # do not want to launch any other nodes before at least one worker
+            # node is connected and its resources show up in LoadMetrics.
+            return_immediately = self._handle_legacy_yaml(
+                nodes, pending_nodes, static_node_resources)
+            if return_immediately:
```

Comment:
Can you give this a better name than return_immediately?

Reply:
Actually how about we rename _handle_legacy_yaml to _infer_legacy_node_resources_if_needed(), remove the early return, and change the logic in line 110 to return empty dict if a worker is already pending?

Reply:
(That way we never need to return early here)

Reply:
@ericl  better?